@here/harp-mapview/lib/MapView.ts

```diff
@@ -2419,18 +2420,15 @@ export class MapView extends THREE.EventDispatcher {
             this.m_forceCameraAspect !== undefined ? this.m_forceCameraAspect : width / height;
         this.setFovOnCamera(this.m_options.fovCalculation!, height);
 
-        // When calculating clip planes account for the highest building on the earth,
-        // multipling its height by projection scalling factor. This approach assumes
-        // constantHeight property of extruded polygon technique is set as default false,
-        // otherwise the near plane margins will be bigger then required, but still correct.
-        const projectionScale = this.projection.getScaleFactor(this.camera.position);
-        const maxHeight = EarthConstants.MAX_BUILDING_HEIGHT * projectionScale;
-        // Copy all properties from new view ranges to our readonly object.
-        // This allows to keep all view ranges references valid and keeps up-to-date
-        // information within them. Works the same as copping all properties one-by-one.
+        const maxGeometryHeight = this.m_tileDataSources.reduce(
+            (r, ds) => Math.max(r, ds.maxGeometryHeight),
+            0
+        );
         Object.assign(
             this.m_viewRanges,
-            viewRanges === undefined ? this.m_visibleTiles.updateClipPlanes(maxHeight) : viewRanges
+            viewRanges === undefined
+                ? this.m_visibleTiles.updateClipPlanes(maxGeometryHeight)
```

Comment:
I don't get this code. We update the clip planes only the first time?

Reply:
I don't get old code too it's old code, but formatted.
By not fixing all the smells i just try to minimize patch and not modify unrelated code. I'll check if this can fixed easily.